### PR TITLE
test: improve test coverage for application and response

### PR DESCRIPTION
**Pull request by fengmk2**
from 98% to 100%

### CODE REVIEW

**Comment by fengmk2** [comment]
app.respond on `null == body` can't be coverage, because `body` never be null or empty string.

```js
  // status body
  if (null == body) {
    this.type = 'text';
    body = this.message || String(code);
    if (body) this.length = Buffer.byteLength(body);
    return res.end(body);
  }
```

**Comment by dougwilson** [comment]
If it can truely never be, then why is the check necessary at all?

**Comment by fengmk2** [comment]
@dead-horse maybe we need to delete `if (body)` detect.

File: lib/application.js [hunk 1]
```diff
@@ -204,6 +204,7 @@ function *respond(next) {
   if (null == body) {
     this.type = 'text';
     body = this.message || String(code);
+    /* istanbul ignore else */
     if (body) this.length = Buffer.byteLength(body);
```
Comment (fengmk2) [inline lib/application.js]: code always be Number, so body never be empty string.

Comment (jonathanong) [inline lib/application.js]: cheater!!! :D

Comment (jonathanong) [inline lib/application.js]: yeah let's delete the `if ()` :+1:

**Comment by fengmk2** [comment]
all done @dougwilson @jonathanong

**Comment by dougwilson** [comment]
LGTM

**Comment by jonathanong** [comment]
LGTM

**Comment by fengmk2** [comment]
Landed 8b14b91bae7fe9302eaefaecbacb67a1f51e494c

**Pull request closed without merging**